scripts/test.sh

```diff
@@ -2,7 +2,7 @@
 
 set -eu
 
-cargo test --all "$@"
+cargo test --all "$@" -- --test-threads=1
```

Comment:
any reason why we need the test thread to be 1 here? Does running them in parallel cause some issues?

Reply:
Ah yeah, I noticed there were some flakey tests due to modifying the environment and colliding with each other.